src/extensions/walls/WallsApplicationCustomizer.ts

```diff
@@ -95,50 +107,105 @@ export default class WallsApplicationCustomizer
+      if (aP.length > 0) aP[0].remove();
+      aP = settingsPane.querySelectorAll("#SuiteMenu_zz8_MenuItemAddPage");
+      if (aP.length > 0) aP[0].remove();
+    }
+
+    //Add app
+    var aP = settingsPane.querySelectorAll('a[href="' + this.context.pageContext.web.serverRelativeUrl + '/_layouts/15/appStore.aspx#myApps?entry=SettingAddAnApp"]');
+    if (aP.length > 0) aP[0].remove();
+    aP = settingsPane.querySelectorAll("#SuiteMenu_zz5_MenuItemCreate");
+    if (aP.length > 0) aP[0].remove();
+
+    //Global Navigation
+    var gN = settingsPane.querySelectorAll('a[href="javascript:_spLaunchGlobalNavSettings();"]');
+    if (gN.length > 0) gN[0].remove();
+    gN = settingsPane.querySelectorAll("#GLOBALNAV_SETTINGS_SUITENAVID");
+    if (gN.length > 0) gN[0].remove();
+
+    //Hub settings
+    var hS = settingsPane.querySelectorAll('a[href="javascript:_spLaunchHubSettings();"]');
+    if (hS.length > 0) hS[0].remove();
+    hS = settingsPane.querySelectorAll("#SUITENAV_HUB_SETTINGS");
+    if (hS.length > 0) hS[0].remove();
+
+    //Site settings
+    var sT = settingsPane.querySelectorAll('a[href="' + this.context.pageContext.web.serverRelativeUrl + '/_layouts/15/settings.aspx"]');
+    if (sT.length > 0) sT[0].remove();
+    sT = settingsPane.querySelectorAll("#SuiteMenu_zz7_MenuItem_Settings");
+    if (sT.length > 0) sT[0].remove();
+
     // Site permissions
     var sP = settingsPane.querySelectorAll('a[href="javascript:_spLaunchSitePermissions();"]');
     if(sP.length > 0) sP[0].remove();
-    sP = settingsPane.querySelectorAll("#SuiteMenu_MenuItem_SitePermissions");
-    if(sP.length > 0) sP[0].remove();
+    sP = settingsPane.querySelectorAll("#SUITENAV_SITE_PERMISSIONS");
+    if (sP.length > 0) sP[0].remove();
 
     // Site information
-    var sI = settingsPane.querySelectorAll('a[href="javascript:_spLaunchSiteSettings();"]');
-    if(sI.length > 0) {
-      let element: HTMLElement = sI[0] as HTMLElement;
-      element.onclick = () => {
-        window.setTimeout(() => {
-          var siteSettingsPane = document.getElementsByClassName("ms-SiteSettingsPanel-SiteInfo");
-          if(siteSettingsPane.length > 0) {
-            window.setTimeout(() => {
-            var jhs = siteSettingsPane[0].getElementsByClassName("ms-SiteSettingsPanel-joinHubSite");
-            if(jhs.length >0 ) jhs[0].remove();
-            }, 300);
-            var c = siteSettingsPane[0].getElementsByClassName("ms-SiteSettingsPanel-classification");
-            if(c.length >0 ) c[0].remove();
-            var p = siteSettingsPane[0].getElementsByClassName("ms-SiteSettingsPanel-PrivacyDropdown");
-            if(p.length >0 ) p[0].remove();
-            var ht = siteSettingsPane[0].getElementsByClassName("ms-SiteSettingsPanel-HelpText");
-            if(ht.length >0 ) ht[0].remove();
-          }
-        }, 500);
+    if (userType === "owner") {
+      var sI = settingsPane.querySelectorAll('a[href="javascript:_spLaunchSiteSettings();"]');
```

Comment:
It appears the Site Contents page doesn't have the same link as the other pages. Will have to figure out a way to try and catch both.